tests/common/fixtures/grpc_fixtures.py

```diff
@@ -0,0 +1,398 @@
+    cert_dir = "/tmp/gnoi_certs"
+    localhost.shell(f"mkdir -p {cert_dir}")
+    localhost.shell(f"cd {cert_dir}")
+
+    # Create Root key
+    localhost.shell(f"cd {cert_dir} && openssl genrsa -out gnmiCA.key 2048")
+
+    # Create Root cert
+    localhost.shell(f"""cd {cert_dir} && openssl req -x509 -new -nodes -key gnmiCA.key -sha256 -days 1825 \
+                       -subj '/CN=test.gnmi.sonic' -out gnmiCA.cer""")
+
+    # Create server key
+    localhost.shell(f"cd {cert_dir} && openssl genrsa -out gnmiserver.key 2048")
+
+    # Create server CSR
+    localhost.shell(f"""cd {cert_dir} && openssl req -new -key gnmiserver.key \
+                       -subj '/CN=test.server.gnmi.sonic' -out gnmiserver.csr""")
+
+    # Create extension file with DUT IP SAN
+    ext_conf_content = f"""[ req_ext ]
+subjectAltName = @alt_names
+[alt_names]
+DNS.1   = hostname.com
+IP      = {duthost.mgmt_ip}"""
+
+    localhost.shell(f"cd {cert_dir} && echo '{ext_conf_content}' > extfile.cnf")
+
+    # Sign server certificate with SAN extension
+    localhost.shell(f"""cd {cert_dir} && openssl x509 -req -in gnmiserver.csr -CA gnmiCA.cer -CAkey gnmiCA.key \
+                       -CAcreateserial -out gnmiserver.cer -days 825 -sha256 \
+                       -extensions req_ext -extfile extfile.cnf""")
+
+    # Create client key
+    localhost.shell(f"cd {cert_dir} && openssl genrsa -out gnmiclient.key 2048")
+
+    # Create client CSR
+    localhost.shell(f"""cd {cert_dir} && openssl req -new -key gnmiclient.key \
+                       -subj '/CN=test.client.gnmi.sonic' -out gnmiclient.csr""")
+
+    # Sign client certificate
+    localhost.shell(f"""cd {cert_dir} && openssl x509 -req -in gnmiclient.csr -CA gnmiCA.cer -CAkey gnmiCA.key \
+                       -CAcreateserial -out gnmiclient.cer -days 825 -sha256""")
+
+    # Copy certificates to DUT
+    duthost.copy(src=f'{cert_dir}/gnmiCA.cer', dest='/etc/sonic/telemetry/')
+    duthost.copy(src=f'{cert_dir}/gnmiserver.cer', dest='/etc/sonic/telemetry/')
+    duthost.copy(src=f'{cert_dir}/gnmiserver.key', dest='/etc/sonic/telemetry/')
+
+    # Copy client certificates to PTF container
+    ptfhost.copy(src=f'{cert_dir}/gnmiCA.cer', dest='/etc/sonic/telemetry/gnmiCA.cer')
+    ptfhost.copy(src=f'{cert_dir}/gnmiclient.cer', dest='/etc/sonic/telemetry/gnmiclient.cer')
+    ptfhost.copy(src=f'{cert_dir}/gnmiclient.key', dest='/etc/sonic/telemetry/gnmiclient.key')
+
+    logger.info("Certificate generation and distribution completed")
+
+
+def _configure_gnoi_tls_server(duthost):
+    """Configure CONFIG_DB for TLS mode."""
+    logger.info("Configuring gNOI server for TLS mode")
+
+    # Configure GNMI table for TLS mode
+    duthost.shell('sonic-db-cli CONFIG_DB hset "GNMI|gnmi" port 50052')
+    duthost.shell('sonic-db-cli CONFIG_DB hset "GNMI|gnmi" client_auth true')
+    duthost.shell('sonic-db-cli CONFIG_DB hset "GNMI|gnmi" log_level 2')
```

Comment:
Where possible, please use `duthost.command` instead of `duthost.shell`. That'll avoid needing to start a shell to run each command and avoids shell interpretation of special characters where that's not desired.